File: back-end/food-care/models/reportModel.js
const mongoose = require("mongoose");

const reportSchema = new mongoose.Schema(
  {
    user_id: {
      type: mongoose.Schema.Types.ObjectId,
      required: true,
      ref: "User",
    },
    post: {
      id: {
        type: String,
      },
      type: {
        type: String,
      },
    },
    description: {
      type: String,
    },
    action: {
      type: String,
    },
  },
  {
    timestamps: true,
  }
);
module.exports = mongoose.model("Report", reportSchema);

File: client/lib/models/reportModel.dart
class Report {
  String? id;
  String? userId;
  String description;
  String action;
  Post post;

  Report({
    this.id,
    this.userId,
    required this.action,
    required this.post,
    required this.description,
  });

  factory Report.fromJson(Map<String, dynamic> json) => Report(
        id: json["_id"],
        userId: json["user_id"],
        description: json["description"],
        action: json["action"],
        post: json["post"] is List
            ? Post.fromJson(json["post"][0])
            : Post.fromJson(json["post"]),
      );

  Map<String, dynamic> toJson() => {
        "_id": id,
        "user_id": userId,
        "post": post.toJson(),
        "description": description,
        "action": action,
      };
}

class Post {
  String id;
  String type;

  Post({
    required this.id,
    required this.type,
  });

  factory Post.fromJson(Map<String, dynamic> json) => Post(
        id: json["id"],
        type: json["type"],
      );

  Map<String, dynamic> toJson() => {
        "id": id,
        "type": type,
      };
}
